File: Mika/branches/mika_classloader/core-vm/java/wonka/vm/JDWP.java
/**************************************************************************
* Copyright (c) 2006 by Chris Gray, /k/ Embedded Java Solutions.          *
* All rights reserved.                                                    *
**************************************************************************/


/*
** $Id: JDWP.java,v 1.1 2006/10/04 14:24:15 cvsroot Exp $
*/

package wonka.vm;

public final class JDWP extends Thread {

  private static JDWP theJDWP;

  private static native boolean isRunning();

  public static synchronized JDWP getInstance() {
    if (theJDWP == null) {
      theJDWP = new JDWP();
    }

    while (!isRunning()) {
      try {
        Thread.sleep(1000);
      }
      catch (InterruptedException ie) {
      }
    }

    return theJDWP;
  }

  private JDWP() {
    super("JDWP");
    setPriority(10);
    setDaemon(true);
    start();
  }

  public native void run();

}

File: Mika/trunk/core-vm/src/native/wonka/vm/JDWP.c
/**************************************************************************
* Copyright (c) 2006 by Chris Gray, /k/ Embedded Java Solutions.          *
* All rights reserved.                                                    *
**************************************************************************/

/*
** $Id: JDWP.c,v 1.1 2006/10/04 14:24:17 cvsroot Exp $
*/

#include "core-classes.h"

#ifdef JDWP
#include "jdwp.h"

w_boolean JDWP_static_isEnabled(JNIEnv *env, w_instance classJDWP) {
  return TRUE;
}

w_boolean JDWP_static_isRunning(JNIEnv *env, w_instance theJDWP) {
  return jdwp_state != jdwp_state_unstarted;
}

void JDWP_run(JNIEnv *env, w_instance theJDWP) {
  jdwp_dispatcher();
}

#else

w_boolean JDWP_static_isEnabled(JNIEnv *env, w_instance classJDWP) {
  return FALSE;
}

w_boolean JDWP_static_isRunning(JNIEnv *env, w_instance theJDWP) {
  // Don't wait before entering main()
  return TRUE;
}

void JDWP_run(JNIEnv *env, w_instance theJDWP) {
  // Just return
}

#endif
